### DISCUSSION TITLE
isEncrypted property

### DISCUSSION THREAD

**Discussion by semproms-64** [Hopding/pdf-lib#1328] [Q&A]
Hello!

As far as I know, PDF-lib can't work with encrypted documents. But I need to work with PDF'S which have the default password (you can open it, but PDF-Lib says they are encrypted and cannot be loaded). This kind of documents can be decrypted with npm libraries such as qpdf. So I'd like to ask if is it possible to detect if a document is encrypted easily so I can use qpdf to decrypt. I always can do something like this:

```
const isEncrypted = async(documentBuffer) => {

  try {
    await PDFDocument.load(documentBuffer);
    return false;
  } catch (e) {
    return String(e).includes('is encrypted');
  }
}
```
But I think this is not the best way to do it. Thank you very much for your help.

**Comment by kenhosr** [comment]
I have a similar question, to be able to decrypt PDFs, especially those with default password. Can someone help?

**Comment by kenhosr** [comment]
https://github.com/Hopding/pdf-lib/pull/1015#issuecomment-984316064 - some related discussion
